src/test/java/org/apache/commons/configuration2/TestDataConfiguration.java

```diff
@@ -73,10 +73,13 @@ private static Date expectedDate() throws ParseException {
 
     /**
      * Create an instance of InternetAddress. This trick is necessary to compile and run the test with Java 1.3 and the
-     * javamail-1.4 which is not compatible with Java 1.3
+     * javamail-1.4 which is not compatible with Java 1.3.
+     *
+     * <p>javamail-2.0 had a namespace change, moving javax.mail.* to jakarta.mail.*. This test verifies if we have
+     * javax.mail.* in the classpath before trying the Jakarta classes.</p>
      */
     private Object createInternetAddress(final String email) throws Exception {
-        final Class<?> cls = Class.forName("javax.mail.internet.InternetAddress");
+        final Class<?> cls = Class.forName("jakarta.mail.internet.InternetAddress");
```

Comment:
In the tests, I think it doesn't make sense to support javax and jakarta, since mailapi is bringing javamail 2.x with the Jakarta namespace.

Reply:
Are we hitting both paths in that `try-catch` during the unit tests?

Reply:
No, only the Jakarta path. I started modifying the test to create a test object using javax.mail too, but then I couldn't find a clear way to load the javax.mail into the classpath. `mailapi` has a different package structure for the mail part, but it also has a common `com.sun.mail` part…

Adding two versions of the `mailapi` results in only the last dependency being added by Maven, and doesn't really sound like a good idea.

Maybe we should leave it as it is until we decide to drop javax.mail and support only jakarta.mail, or leave it untested? Or maybe there's some other way to test with different dependencies (never had to do that). WDYT?

![image](https://user-images.githubusercontent.com/304786/167767328-8638a868-6cc9-41cd-b408-ee637a0aabd3.png)

Reply:
Let's resolve this one way or the other before we cut a release candidate... @kinow any thoughts?

Reply:
I think this doesn't break binary backward compatibility, and doesn't add anything to the public API. It allows users to switch from `javax.mail` to `jakarta.mail` by simply changing the API implementation in their classpath. Theoretically both should work fine, but I don't know how to test both scenarios in Commons Configuration (so the test is validating only the newer `jakarta.mail` since this PR bumps 1.6.6. to 2.0.1).

So I think it should be OK to include this change in the next release.

Reply:
I just tested this by swapping the mail api version back to 1.6.7 and running the tests. Everything passes on the old and new versions if we switch the catch to a `ClassNotFoundException` as I suggested above.